compiler/env/jittypes.h

```diff
@@ -149,11 +149,8 @@ typedef struct TR_InlinedCallSite
       #define FLUSH_MEMORY(smp)
    #endif
 #else
-   #if (__IBMC__ || __IBMCPP__)
-      void do_lwsync (void);
-      #pragma mc_func do_lwsync {"7c2004ac"}
-      #pragma reg_killed_by do_lwsync
-      #define FLUSH_MEMORY(smp) if( smp ) do_lwsync();
+   #if defined(__IBMC__) || defined(__IBMCPP__)
+      #define FLUSH_MEMORY(smp) if( smp ) __lwsync();
    #elif defined(LINUX)
       #define FLUSH_MEMORY(smp) if( smp ) __asm__("sync");
```

Comment:
This has nothing to do with XLC, but as long as you're making changes in this code I think we can change this "sync" to "lwsync" to keep the behaviour consistent across all compilers.